docs/user-guide.md

```diff
@@ -129,6 +129,44 @@ After running `kubectl cluster-compare` tool, the reference configuration CRs wi
 
 ## Options and advanced usage
 
+### Diff config
+
+The command includes an option to pass a config that includes user specified settings regarding on how to do the diff.
+The config is formatted as a YAML file.
+
+#### Manual Correlation
+
+There is an option to select for specific CRs the template that it will be diffed with, this will overwrite the default
+correlating.
+The matches between the cluster CRs and the templates can be added to the diff config as pairs of `apiVersion_kind_namespace_name:
+<Template File Name>`. For cluster scoped CRs that don't have a namespace the matches can be added as pairs of
+`apiVersion_kind_name: <Template File Name>`.
+For example:
+
+```yaml
+correlationSettings:
+   manualCorrelation:
+      correlationPairs:
+         apps.v1.DaemonSet.kube-system.kindnet.yaml: "template_example.yaml"
+```
+
+#### Specifying Template match choosing mechanism in case of two templates with same group
+
+In cases where the reference includes two templates with the same apiVersion-kind-namespace-name 4-tuple, by the correlation
+logic that was explained in previous section the command will diff with the template that is best matched to the CR.
+The best match behaviour can be configured in the diff config. The default behaviour is to use the template with the least
+number of lines in the diff with the CR (Named 'NumLines'). Is also possible to disable this mechanism by specifying 'None'
+under the bestMatchType section in the user config. The only supported values are: NumLines and None. More mechanisms can
+be added by request.
+
+```yaml
+bestMatchType: "None"
```

Comment:
Sorry I missed this last time I looked. This seems more like a choice that should be made my the reference designer than the end user.

Reply:
i think its more of a user decision because it depends on the diff tool used by the user that is passed in the env var.

Reply:
Perhaps how to calculate the number of diffs is diff to dependent.

If there are templates which need this is reference implementation decision. 

Perhaps using something like generating a merge patch and counting the number of leaf nodes in the yaml would be more tool agnostic metric.